File: Solutions/3075- Maximize Happiness of Selected Children/Solution.java
class Solution {
    public long maximumHappinessSum(int[] happiness, int k) {
        long answer = 0;
        Arrays.sort(happiness);

        for (int i = 0; i < k; i++) {
            int temp = Math.max(happiness[happiness.length - 1 - i] - i,0);
            answer += temp;
        }

        return answer;
    }
}

File: Solutions/3075- Maximize Happiness of Selected Children/Solution.py
class Solution:
    def maximumHappinessSum(self, happiness: List[int], k: int) -> int:
        happiness.sort()
        answer = 0

        for i in range(0,k):
            temp = max(happiness[-1 - i] - i,0)
            answer += temp

        return answer